src/webfaf2/problems.py

```diff
@@ -151,6 +153,30 @@ def query_problems(db, hist_table, hist_column,
 
         final_query = final_query.filter(Problem.id == names_query.c.problem_id)
 
+    version_query = (
+        db.session.query(Report.problem_id.label("problem_id"))
+        .join(ReportPackage)
+        .join(Package)
+        .join(Build)
+        .filter(ReportPackage.type == "CRASHED")
+        .distinct(Report.problem_id))
+
+    if since_version:
+        ver_sq = version_query.filter(Build.semver >= since_version).subquery()
+        final_query = final_query.filter(Problem.id == ver_sq.c.problem_id)
+
+    if since_release:
+        ver_sq = version_query.filter(Build.semrel >= since_release).subquery()
+        final_query = final_query.filter(Problem.id == ver_sq.c.problem_id)
+
+    if to_version:
+        ver_sq = version_query.filter(Build.semver <= to_version).subquery()
+        final_query = final_query.filter(Problem.id == ver_sq.c.problem_id)
+
+    if to_release:
+        ver_sq = version_query.filter(Build.semrel <= to_release).subquery()
+        final_query = final_query.filter(Problem.id == ver_sq.c.problem_id)
+
```

Comment:
This requires both the version AND release to be higher than specified. If we have two problems with package versions 3.10.0-10 and 3.11.0-5, then searching this way for newer than 3.10.0-9 doesn't yield the 3.11.0-5, because it's release is less than 9, even though the more significant version is higher. The correct way was outlined here https://github.com/abrt/faf/pull/390#issuecomment-76385903 .

Also this creates a whole new subquery for every parameter used, which seems costly. I'd rather put all the filters into one subquery and apply it only once to the `final_query`, as is the case with the `function_names` etc. filter just above.